src/core/border_router/routing_manager.hpp

```diff
@@ -154,36 +154,52 @@ class RoutingManager : public InstanceLocator
         kOnLinkPrefixLength     = OT_IP6_PREFIX_BITSIZE, // The length of an On-link prefix. In bits.
     };
 
+    enum : uint32_t
+    {
+        kMaxInitRtrAdvertisements = 3, // The maximum number of initial Router Advertisements.
+        kMaxRtrSolicitations = 3, // The Maximum number of Router Solicitations before sending Router Advertisements.
+    };
+
     enum : uint32_t
     {
         kDefaultOmrPrefixLifetime    = 1800u,                  // The default OMR prefix valid lifetime. In seconds.
         kDefaultOnLinkPrefixLifetime = 1800u,                  // The default on-link prefix valid lifetime. In seconds.
         kMaxRtrAdvInterval           = 600,                    // Maximum Router Advertisement Interval. In seconds.
         kMinRtrAdvInterval           = kMaxRtrAdvInterval / 3, // Minimum Router Advertisement Interval. In seconds.
-        kMaxInitRtrAdvInterval       = 16,  // Maximum Initial Router Advertisement Interval. In seconds.
-        kMaxRaDelayTime              = 500, // The maximum delay of sending RA after receiving RS. In milliseconds.
-        kRtrSolicitationInterval     = 4,   // The interval between Router Solicitations. In seconds.
-        kMaxRtrSolicitationDelay     = 1,   // The maximum delay for initial solicitation. In seconds.
-        kMaxRoutingPolicyDelay       = 1,   // The maximum delay for routing policy evaluation. In seconds.
+        kMaxInitRtrAdvInterval       = 16,   // Maximum Initial Router Advertisement Interval. In seconds.
+        kMaxRaDelayTime              = 500,  // The maximum delay of sending RA after receiving RS. In milliseconds.
+        kRtrSolicitationInterval     = 4,    // The interval between Router Solicitations. In seconds.
+        kMaxRtrSolicitationDelay     = 1,    // The maximum delay for initial solicitation. In seconds.
+        kMaxRoutingPolicyDelay       = 1,    // The maximum delay for routing policy evaluation. In seconds.
+        kRtrAdvStaleTime             = 1500, // The period of Router Solicitations. In seconds.
```

Comment:
This is not really the period of RS messages. In particular, when there are multiple on-link prefixes, we should enter router discover after the last non-stale on-link prefix becomes stale.